runtime/parachains/src/paras_inherent.rs

```diff
@@ -166,20 +325,29 @@ pub mod pallet {
 				parent_header,
 				disputes,
 			} = data;
+			let total_weight =
+				paras_inherent_total_weight::<T>(&backed_candidates, &signed_bitfields, &disputes);
+
+			// Abort if the total weight of the block exceeds the max block weight
+			ensure!(
+				total_weight <= <T as frame_system::Config>::BlockWeights::get().max_block,
+				Error::<T>::InherentOverweight
+			);
```

Comment:
> pick randomly from local disputes

we're using specific disputes ordering when handling them in #4134, it should be the same here `(relay_parent_number, candidate_hash)`

Reply:
I haven't read that PR, but how would we enforce ordering from the node in the runtime? Malicious block author can always pass data with bad ordering

Reply:
To my understanding, it's up to the block producer to include the votes. And hence, if a an overweight block hits the `fn ParaInherent::enter` we must choose randomly. For the `fn create_inherent`, we might prefer the first `n` ones instead of random selection.

Reply:
Talking to Robert, we want to sort, and only taking local ones is fine. We should prefer older over newer dispute messages.